flake.nix

```diff
@@ -4,22 +4,37 @@
 
   outputs = { self, nixpkgs }: {
     lib = {
-      # TODO handle multiple architectures.
-      evalConfig = import ./eval-config.nix { inherit (nixpkgs) lib; };
-
-      darwinSystem =
-        { modules, inputs ? { }
-        , system ? throw "darwin.lib.darwinSystem now requires 'system' to be passed explicitly"
-        , ...
-        }@args:
-        self.lib.evalConfig (args // {
-          inherit system;
-          inputs = { inherit nixpkgs; darwin = self; } // inputs;
-          modules = modules ++ [ self.darwinModules.flakeOverrides ];
-        });
+      evalConfig = import ./eval-config.nix;
+
+      darwinSystem = args@{ modules, ... }: self.lib.evalConfig (
+        { inherit (nixpkgs) lib; }
+        // nixpkgs.lib.optionalAttrs (args ? pkgs) { inherit (args.pkgs) lib; }
+        // builtins.removeAttrs args [ "system" "pkgs" "inputs" ]
+        // {
+          modules = modules
+            ++ nixpkgs.lib.optional (args ? pkgs) ({ lib, ... }: {
+              _module.args.pkgs = lib.mkForce args.pkgs;
+            })
+            # Backwards compatibility shim; TODO: warn?
+            ++ nixpkgs.lib.optional (args ? system) ({ lib, ... }: {
+              nixpkgs.system = lib.mkDefault args.system;
+            })
+            # Backwards compatibility shim; TODO: warn?
+            ++ nixpkgs.lib.optional (args ? inputs) {
+              _module.args.inputs = args.inputs;
+            }
+            ++ [ ({ lib, ... }: {
+              nixpkgs.source = lib.mkDefault nixpkgs;
+
+              system.checks.verifyNixPath = lib.mkDefault false;
+              system.checks.verifyNixChannels = lib.mkDefault false;
```

Comment:
Sure, I can remove that line.

Reply:
Maybe the check isn't useful, but I feel like being able to run eg. `nix-build '<nixpkgs>' -A hello` or a file with `import <nixpkgs> {}` is somewhat part of a working installation.

Reply:
Did you mean both these checks or only the latter? `verifyNixChannels` seems harmless to me, since it looks like it only fails if you have *broken* channels rather than no channels at all. `verifyNixPath` I think would be a regression for flake setups, since it requires `darwin-config` which doesn't really make sense in that context. (I personally keep a completely blank Nix path and use `NIX_PATH=nixpkgs=flake:nixpkgs` when I have to.)

Reply:
Yeah, there's more to say for not verifying NIX_PATH. I meant the broken channels specifically.